library/SmartyPlugins/function.asset.php

```diff
@@ -17,31 +17,31 @@
  * @param object $smarty Smarty The smarty object rendering the template.
  * @return string The rendered asset.
  */
-function smarty_function_asset($params, &$smarty) {
-    $name = val('name', $params);
-	$tag = val('tag', $params, '');
-	$id = val('id', $params, $name);
-
-	$class = val('class', $params, '');
-	if ($class != '') {
-		$class = ' class="'.$class.'"';
+function smarty_function_asset(array $params, &$smarty) {
```

Comment:
I have high confidence in this change not breaking anything because smarty prepares this data and passes an array.

Reply:
I'm not keen on applying type restrictions on publicly-accessible methods/functions, especially ones as old as this. *However*, I concede this will likely not have an adverse effect for the reasons already stated.